src/robot/libraries/Collections.py

```diff
@@ -432,6 +454,13 @@ def _index_error(self, list_, index):
         raise IndexError('Given index %s is out of the range 0-%d.'
                          % (index, len(list_)-1))
 
+    def _validate_list(self, list, position=1):
+        if not is_list_like(list):
+            raise TypeError("Expected %i. argument to be a list, got '%s' instead." % (position, type_name(list)))
```

Comment:
Nitpicky style comments:
- It would be better to use `list_`, not `list` as the argument name.
- `%d` is used a lot more commonly than `%i` in Python. Didn't myself even know what `%i` does without googling.
- I don't think `1.` style is used in English and it should be `Expected 1st argument to be...` instead. A lot easier to change it to `Expected argument 1 to be...`, though.
- We don't expect the argument to be a list, only list-like, so the message should be something like `Expected argument %d to be list or list-like, ...`.
- Could actually consider renaming this helper to `_validate_list_like` as well.
- Quotes aren't needed around the name returned by `type_name`. The message should thus end just `..., got %s instead.`
- We try to follow PEP-8 and keep lines below 80 characters.

This implementation ought to fix all the (minor) problems listed above:

```python
    def _validate_list(self, list_, position=1):
        if not is_list_like(list_):
            raise TypeError("Expected argument %d to be list or list-like, "
                            "got %s instead." % (position, type_name(list_)))
```